src/CornerstoneViewport/CornerstoneViewport.js

```diff
@@ -388,9 +403,14 @@ class CornerstoneViewport extends Component {
       this.onNewImage
     );
 
-    // Update our "Images Loaded" count.
-    // Better than nothing?
-    this.element[addOrRemoveEventListener](
+    // Update image load progress
+    cornerstone.events[addOrRemoveEventListener](
+      'cornerstoneimageloadprogress',
+      this.onImageProgress
+    );
```

Comment:
I'm pretty sure this is the load progress for each image XHR request. At least in `cornerstone-wado-image-loader`. If that's true, we probably want `onImageLoadProgress` to only update `imageProgress` if it's for the `imageId` at the index we're waiting on.

I don't think this is a blocker, but it's food for thought and probably worth a follow-up enhancement label if you don't want to tackle it this go-round.

Reply:
I think it already does that. I have not verified it though. Also, you are right about not having enough bandwidth right now.